### ISSUE TITLE
Feature Request: Support to read and write raw content (without any formatting) from and into clipboards

### ISSUE THREAD

**Issue by rams3sh** [golang-design/clipboard#40]
Hey 

First of all thanks a ton for this project. I have been experimenting with the project for a week now. I have a requirement for an internal project where I have to detect the content type of the clipboard as file or text and take an action based on that. 

I tried the clipboard package and when I try to copy a folder or a file (agnostic of any format) , the `Read` function gives only the text path of the folder. Hence I am unable to differentiate if the copied content is just the text file path or the actual file itself. 

My request pertains to  providing an additional method to directly read the raw content from the clipboard (hex format / any other suitable binary format) , than forcing it to be formatted to a Text or Image (which is the case as on date) in which case it loses context of the content; and a direct write of raw content to the clipboard.

This will also help others to write their own implementation of formatting until the support for their requested format is added to the main library.

**Comment by changkun** [comment]
This is subsumed by the custom-format design tracked in #17 (spec proposed in #101).

Custom formats are defined as **raw byte passthrough** — no forced text/image conversion — which is exactly the raw read/write requested here. Once it lands you'll be able to do:

```go
f := clipboard.Register("application/octet-stream") // or any native type
clipboard.Write(f, raw)
b := clipboard.Read(f)
```

and read a platform-native type (e.g. the file-list type) to distinguish a copied file from its text path, rather than being forced into `FmtText`/`FmtImage`.

Closing in favour of #17; please follow along there. Refs #17, #101.

**Comment by changkun** [comment]
Shipped via the custom-format support (#17): `clipboard.Register(mime)` returns a `Format` token whose `Read`/`Write` move the exact bytes with **no** conversion — i.e. raw read/write for any MIME type (including `application/octet-stream`). Available on all desktop platforms.
